Add new profile: nicotine

chrpinedo opened the pull request:
New profile for Nicotine Plus Soulseek client. 
I realized you were asking contribution for another Soulseek client (soulseekqt), but this is the one I use.
The profile is heavily based on Amule profile and I tried to follow your contribution guidelines.

Tested in Arch Linux with Nicotine Plus 1.4.1-4 and Firejail 0.9.62-1.

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+seccomp
+shell none
+
+private-bin nicotine,python2
```
```suggestion
private-bin nicotine,python2*
```

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
```
what about `tracelog`?

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
```
```suggestion
include disable-programs.inc
include disable-xdg.inc
```

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+mkdir ${HOME}/.nicotine
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/.nicotine
+include whitelist-common.inc
```
What about `include whitelist-runuser-common.inc`., `include whitelist-usr-share-common.inc` and `include whitelist-var-common.inc`?

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+
+apparmor
+caps.drop all
+ipc-namespace
```
```suggestion
#ipc-namespace
```

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+protocol unix,inet,inet6
+seccomp
+shell none
+
```
Can we add `disable-mnt`.

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+shell none
+
+private-bin nicotine,python2
+private-dev
```
What about `private-cache`?

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,44 @@
+
+private-bin nicotine,python2
+private-dev
+private-tmp
```
Does it need D-Bus or can we add `dbus-user none` and `dbus-system none`?
To test:
Both none: `nodbus`
only system none: `blacklist /run/dbus`

chrpinedo commented:
I added your recommendations in the last commit. The application works.

The only thing that doesn't work is the "open directory" context menu of a downloaded file, which opens the directory where the file was downloaded to with the default file manager. To allow this functionality I should allow more binaries in the `private-bin` statement such as xdg-mime, xdg-open and a set of file manager such as GNOME Nautilus, Thunar... I think it is not an important feature and it is more important to harden this old application. What do you think?

rusty-snake commented on etc/nicotine.profile:
```diff
@@ -0,0 +1,53 @@
+#ipc-namespace
+netfilter
+no3d
+nodbus
```
Remove `nodbus` here and add `dbus-user none` & `dbus-system none` in a new block to the end of the profile.

chrpinedo commented:
I removed `nodbus` and added `dbus-user none` and `dbus-system none` sentences.

I also rebase the three commits related to `nicotine.profile` file in one unique commit for clarity of the merge.

rusty-snake commented:
Merged, Thanks.

captaidman commented:
Linux Mint 19.3

`$ nicotine
Reading profile /usr/local/etc/firejail/nicotine.profile
Reading profile /usr/local/etc/firejail/allow-python2.inc
Reading profile /usr/local/etc/firejail/disable-common.inc
Reading profile /usr/local/etc/firejail/disable-devel.inc
Reading profile /usr/local/etc/firejail/disable-exec.inc
Reading profile /usr/local/etc/firejail/disable-interpreters.inc
Reading profile /usr/local/etc/firejail/disable-passwdmgr.inc
Reading profile /usr/local/etc/firejail/disable-programs.inc
Reading profile /usr/local/etc/firejail/disable-xdg.inc
Reading profile /usr/local/etc/firejail/whitelist-common.inc
Reading profile /usr/local/etc/firejail/whitelist-runuser-common.inc
Reading profile /usr/local/etc/firejail/whitelist-usr-share-common.inc
Reading profile /usr/local/etc/firejail/whitelist-var-common.inc
Parent pid 15275, child pid 15276
8 programs installed in 22.20 ms
Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Blacklist violations are logged to syslog
Child process initialized in 157.77 ms

Parent is shutting down, bye...`

chrpinedo commented:
Arch Linux and it works for me.

```bash
$ firejail --version
firejail version 0.9.62

Compile time support:
	- AppArmor support is enabled
	- AppImage support is enabled
	- chroot support is enabled
	- file and directory whitelisting support is enabled
	- file transfer support is enabled
	- firetunnel support is enabled
	- networking support is enabled
	- overlayfs support is enabled
	- private-home support is enabled
	- seccomp-bpf support is enabled
	- user namespace support is enabled
	- X11 sandboxing support is enabled
```


```bash
$ nicotine 
Reading profile /home/zako/.config/firejail/nicotine.profile
Reading profile /etc/firejail/allow-python2.inc
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/disable-xdg.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /home/zako/.config/firejail/whitelist-runuser-common.inc
Reading profile /etc/firejail/whitelist-usr-share-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 8471, child pid 8472
41 programs installed in 42.94 ms
Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Warning: cleaning all supplementary groups
Warning: /sbin directory link was not blacklisted
Warning: /usr/sbin directory link was not blacklisted
Blacklist violations are logged to syslog
Child process initialized in 125.34 ms
```
